slam-libraries/viam-orb-slam3/orbslam_server_v1.cc

```diff
@@ -22,9 +25,43 @@ using viam::common::v1::PoseInFrame;
 #define MAX_COLOR_VALUE 255
 const std::string strRGB = "/rgb";
 const std::string strDepth = "/depth";
+const auto HEADERTEMPLATE =
+    "VERSION .7\n"
+    "FIELDS x y z\n"
+    // NOTE: If a float is more than 4 bytes
+    // on a given platform
+    // this size will be inaccurate
+    "SIZE 4 4 4\n"
+    "TYPE F F F\n"
+    "COUNT 1 1 1\n"
+    "WIDTH %d\n"
+    "HEIGHT 1\n"
+    "VIEWPOINT 0 0 0 1 0 0 0\n"
+    "POINTS %d\n"
+    "DATA binary\n";
 
 namespace viam {
 
+/*
+applies the mapSize to the HEADERTEMPLATE
+returning the pcd header as a string
+*/
+std::string pcdHeader(int mapSize) {
+    return str(boost::format(HEADERTEMPLATE) % mapSize % mapSize);
+}
+
+/*
+casts the float f to a pointer of unsigned 8 bit bytes
+iterates throught all the 8 bit bytes of f
+writes each 8 bit bytes to buffer
+*/
+void writeFloatToBufferInBytes(std::string &buffer, float f) {
```

Comment:
I'm passing buffer as a reference as I'm under the impression (from the tour of C++ book) that passing references rather than pointers is preferable whenever possible, which it is in this case.  Please let me know if that is not the case.

Reply:
I've also seen the pattern of passing a const reference (for things that can be const) and a non-const pointer (for things that cannot be const). I'm not sure why people use pointers for non-const sometimes. But I think what you've done here makes sense.